### ISSUE TITLE
Provide the ability to sort implemented/re-implemented list

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10776]
**Describe the bug**
The implemented/re-implemented list is not sorted. See the following testcase

```
/**
 * Base docs
 */
class Base {
public:
    /**
     * fun docs
     */
    virtual void fun() = 0;
};

/**
 * C docs
 */
class C : public Base {
public:
    /**
     * fun docs
     */
    virtual void fun();
};

/**
 * A docs
 */
class A : public Base {
public:
    /**
     * fun docs
     */
    virtual void fun();
};

/**
 * B docs
 */
class B : public Base {
public:
    /**
     * fun docs
     */
    virtual void fun();
};
```

The list shows `Implemented in C, A and B`.  It looks like the list is populated based on when the class is parsed.  If the derived classes are implemented across multiple files, the order could be different depending on when the files are first parsed. It would be helpful to provide the ability to sort the lists to allow simpler diff between two different versions of Doxygen. 

**Screenshots**
![image](https://github.com/doxygen/doxygen/assets/32584126/c6073fcf-24d5-4201-93fe-22892fa40edf)

**To Reproduce**
[test_doxygen.zip](https://github.com/doxygen/doxygen/files/14828190/test_doxygen.zip)

**Expected behavior**
I would like to have list ordered, i.e., `Implemented in A, B and C`.

**Version**
1.8.9.1 - 1.10.0, also on trunk (cbb48edb43715e3420a69d69c21e6e3c4f998894)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10779

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
